### DISCUSSION TITLE
USB slave and DIN midi

### DISCUSSION THREAD

**Discussion by Keeze** [probonopd/MiniDexed#452] [Q&A]
Hi,

I just breadboarded my Minidexed and love it! So now it is time to create a final product. I would like to have the posibility to be able to use both USB (client, to PC) midi in and good old fashioned DIN midi in. 

Has someone implemented this? I am thinking to make use of a Pi Pico or Teensy LC and connect the DIN midi to one of these boards and let the Pico/LC handle the incoming mididata.

Note: I want to be able to choose between USB and DIN, not use them at the same time.

Cheers,
Kees

**Comment by probonopd** [comment]
You can use either Serial MIDI or USB MIDI in by changing the configuration file. No need for a Pi Pico or Teensy LC unless I am overlooking some requirement of yours.

**Reply by Keeze** [reply]
I want to be able to connect the Minidexed to my PC and to devices using DIN midi. So the USB must be client....

**Reply by probonopd** [reply]
I see. Yes, then the way you suggest will work.

**Reply by probonopd** [reply]
Have a look at https://github.com/rsta2/pico/tree/main/midi_adapter, it might serve your needs.

**Comment by miotislucifugis** [comment]
I think that i may have requested this before, but it would make a lot of sense if the midi configuration was an editable and savable "performance" feature.   Minidexed is actually a pretty capable midi utility- can even be used as a stand-alone  usb midi host...   I could definitely see how a user (other than myself) would want to have the thru routing changeable for different performance settings.... without having to eject the card and edit  minidexed.ini

**Reply by donluca** [reply]
I'm actually seconding this, it could be really useful, maybe you might want to open a separate thread with a feature request for better visibility.
